src/ServiceBus/ServiceBus/Cmdlets/NetworkRuleSet/AddAzureServicebusIPRule.cs

```diff
@@ -0,0 +1,88 @@
+    {
+        [Parameter(Mandatory = true, ParameterSetName = IPRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [Parameter(Mandatory = true, ParameterSetName = IPRuleInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = IPRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [Parameter(Mandatory = true, ParameterSetName = IPRuleInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Name { get; set; }        
+
+        [Parameter(Mandatory = true, ParameterSetName = IPRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "Subnet Resource ID")]
+        [ValidateNotNullOrEmpty]
+        public string IpMask { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = IPRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "The IP Filter Action")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("Allow")]
+        public string Action { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = IPRuleInputObjectParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "IPRule Configuration Object to be added")]
+        [ValidateNotNullOrEmpty]
+        public PSNWRuleSetIpRulesAttributes IpRuleObject { get; set; }
+
+
+        public override void ExecuteCmdlet()
+        {
+            PSNetworkRuleSetAttributes networkRuleSet = Client.GetNetworkRuleSet(ResourceGroupName, Name);
+            if (!networkRuleSet.IpRules.Contains(new PSNWRuleSetIpRulesAttributes { IpMask = IpMask }))
+            {
+                if (ShouldProcess(target: Name, action: string.Format("Adding IPrule for NetworkRuleSet of {0} in Resourcegroup {1}", Name, ResourceGroupName)))
+                {
+                    try
+                    {
+                        //Add the IpRules
+                        if (ParameterSetName.Equals(IPRulePropertiesParameterSet))
+                        {
+                            networkRuleSet.IpRules.Add(new PSNWRuleSetIpRulesAttributes { IpMask = IpMask, Action = "Allow" });
+                        }
+
+                        if (ParameterSetName.Equals(IPRuleInputObjectParameterSet))
+                        {
+                            networkRuleSet.IpRules.Add(IpRuleObject);
+                        }
+
+                        Client.CreateOrUpdateNetworkRuleSet(ResourceGroupName, Name, networkRuleSet);
```

Comment:
@v-Ajnava same comment here -- it doesn't look like this cmdlet is writing anything to the output stream